### PR TITLE
Videos Page

### PR DESCRIPTION
**Pull request by foolhardy21**
## Videos Page.

- This PR has the components, filtering logic and APIs required on video listing page.

- Files to review:
   - [ ] src/contexts/filter.context.js
   - [ ] src/contexts/videos.context.js
   - [ ] src/pages/Videos.js
   - [ ] src/components/Videos/VideosSection.js
   - [ ] src/components/Videos/VideosFilter.js
   - [ ] src/components/Videos/VideoCard.js

- Use this link for preview -  https://deploy-preview-3--pensive-hodgkin-f2cf81.netlify.app/videos

v-1

Thanks.

### CODE REVIEW

File: src/components/Videos/VideoCard.js [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+    video,
+    videoTitle,
+    videoDescription
+} }) => {
+    const { theme } = useTheme()
+
+    return (
+        <Card id='container-video' classes='pd-xs pos-relative'>
+
+            {/* <Button classes={`btn-solid ${getSolidBtnBgColor(theme)} ${getSolidBtnTextColor(theme)} txt-md txt-lcase pd-xs pos-absolute tr-1`}>watch later</Button> */}
```
Comment (b0llu) [inline src/components/Videos/VideoCard.js]: remove the commented button

**Review by b0llu — APPROVED** [review]
Suggested one small change everything else looks good

File: src/pages/Videos.js [hunk 2]
```diff
@@ -41,7 +42,18 @@ const Videos = () => {
...
+                {
+                    type === 'error'
+                        ? <Alert classes='bg-err'>{message}</Alert>
+                        : type === 'success' ? <Alert classes='bg-success'>{message}</Alert>
+                            : ''
+                }
+
+                {
+                    loading
+                        ? <Text classes={`${getTextColor(theme)} txt-xlg txt-500 txt-cap mg-top-md`}>loading...</Text>
```
Comment (MayankKumar10) [inline src/pages/Videos.js:54]: you can also use loader package for loading here

**Review by MayankKumar10 — APPROVED** [review]
Just suggesting one thing everything else looks good to me you can merge it

File: src/components/Videos/VideoCard.js (hunk 1 above)
Comment (foolhardy21) [reply src/components/Videos/VideoCard.js]: thanks for pointing out. removed.

File: src/pages/Videos.js (hunk 2 above)
Comment (foolhardy21) [reply src/pages/Videos.js:54]: I'll look into it definitely.

**Pull request merged**